operator/v1/types_csi_cluster_driver.go

```diff
@@ -96,6 +96,44 @@ type ClusterCSIDriverSpec struct {
 	// The current default behaviour is Managed.
 	// +optional
 	StorageClassState StorageClassStateName `json:"storageClassState,omitempty"`
+
+	// DriverConfig field can be used to optionally specify
+	// driver specific configuration.
+	// If this field is nil, platform chosen default configuration
+	// will be applied.
+	// +kubebuilder:validation:Optional
+	// +optional
+	DriverConfig *CSIDriverConfigSpec `json:"driverConfig,omitempty"`
+}
+
+// CSIDriverConfigSpec defines configuration spec that can be
+// used to optionally configure a specific CSI Driver.
+// +union
+type CSIDriverConfigSpec struct {
```

Comment:
I would vote for this API to be platform-agnostic. I think we should consider using a `runtime.RawExtension` here. 

For example, in order to backport a CSI driver configuration to a previous OCP version we would have to backport the API as well. I definitely can see this happening in the future.

Reply:
Not sure I understand. Are we going to backport topology support to 4.11 and older versions of OCP? It could be tricky. Do we have a concrete case where this may happen?

Reply:
Actually I'm talking about other configurations (vSphere or not) that need to be backported to previous releases in order to fix bugs.

For example, in OCP 4.14 we fix a bug in GCP Filestore that require the user to provide a `clustercsidriver.Spec.Config.NetworkName` field. Then someone asks us to backport that fix to OCP 4.13. If `Config` field is hardcoded type in the API, we will need to backport the API as well. If it's a `runtime.RawExtension`, we don't need to backport the API changes (just the code in the operator).

Do you think this is something that can happen?

BTW, this has happened to me with proxy support. Because `ObservedConfig` was a [raw field](https://github.com/bertinatto/library-go/blob/fef421ec24f46a3abfb64fab52a52585059329f2/vendor/github.com/openshift/api/operator/v1/types.go#L88), we didn't have to backport the API changes.

Reply:
From user perspective it does not really matter if an API is provided by `RawExtension` or a real structs / fields. It's still an API and they need to know what yaml / json / protobuf files they can use. And we don't backport new APIs, regardless how they are implemented internally.

Therefore I am for proper structs + fields, as they are more explicit.

`ObservedConfig` was technically backport of a new API that nobody noticed.